### PR TITLE
Add mqtt hook for WebPubSub.AspNetCore project

### PR DESCRIPTION
**Pull request by Y-Sindo**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs [hunk 1]
```diff
@@ -30,6 +30,17 @@ public virtual ValueTask<ConnectEventResponse> OnConnectAsync(ConnectEventReques
             return default;
         }
 
+        /// <summary>
+        /// Mqtt connect event method.
+        /// </summary>
+        /// <param name="request"></param>
```
Comment (JialinXin) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs:36]: do not leave empty.

**Review by JialinXin — COMMENTED** [review]
Please also update generated API and Readme.

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/Shared/MqttConnectCodeToHttpStatusCodeConverter.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                MqttV500ConnectReasonCode.TopicNameInvalid => HttpStatusCode.BadRequest,
+                MqttV500ConnectReasonCode.PayloadFormatInvalid => HttpStatusCode.BadRequest,
+                MqttV500ConnectReasonCode.ImplementationSpecificError => HttpStatusCode.BadRequest,
+                MqttV500ConnectReasonCode.PacketTooLarge => HttpStatusCode.BadRequest,
+                MqttV500ConnectReasonCode.RetainNotSupported => HttpStatusCode.BadRequest,
+                MqttV500ConnectReasonCode.QosNotSupported => HttpStatusCode.BadRequest,
+
+                // Too many requests
+                MqttV500ConnectReasonCode.QuotaExceeded => (HttpStatusCode)429,
+                MqttV500ConnectReasonCode.ConnectionRateExceeded => (HttpStatusCode)429,
```
[the remark this answers is not in the document]
Comment (Y-Sindo) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/Shared/MqttConnectCodeToHttpStatusCodeConverter.cs:47]: There isn't such an enum defined in `HttpStatusCode`

[the remark this answers is not in the document]
Comment (JialinXin) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/src/Shared/MqttConnectCodeToHttpStatusCodeConverter.cs:47]: Oh I see. It's available in System.Net 8.0

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/CHANGELOG.md [hunk 3]
```diff
@@ -1,14 +1,9 @@
...
+## 1.4.0 (2024-09-10)
 
 ### Features Added
-
-### Breaking Changes
-
-### Bugs Fixed
-
-### Other Changes
+- Supported MQTT client event handling.
```
Comment (JialinXin) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/CHANGELOG.md:6]: `Support` ?

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs [hunk 4]
```diff
@@ -30,6 +30,21 @@ public virtual ValueTask<ConnectEventResponse> OnConnectAsync(ConnectEventReques
...
+        /// <summary>
+        /// MQTT connect event method.
+        /// </summary>
+        /// <param name="request"><see cref="MqttConnectEventRequest"/> to get client connect request information.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> used to cancel the request.</param>
+        /// <returns>
+        /// <para>If you want to accept the connection, return an <see cref="MqttConnectEventRequest.CreateMqttResponse(string, System.Collections.Generic.IEnumerable{string}, System.Collections.Generic.IEnumerable{string})"/>.</para>
+        /// <para>If you want to reject the connection, return an <see cref="MqttConnectEventRequest.CreateMqttV311ErrorResponse(MqttV311ConnectReturnCode, string?)"/> or <see cref="MqttConnectEventRequest.CreateMqttV50ErrorResponse(MqttV500ConnectReasonCode, string?)"/></para>
+        /// </returns>
+        /// <remarks>If you don't override this method, MQTT "connect" events are handled by <see cref="OnConnectAsync(ConnectEventRequest, CancellationToken)"/> method.</remarks>
```
Comment (JialinXin) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs:42]: typo here: `OnMqttConnectAsync(MqttConnectEventRequest request, CancellationToken cancellationToken)`, or maybe simply explain as `are handled by default return`?

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Microsoft.Azure.WebPubSub.AspNetCore.csproj [hunk 5]
```diff
@@ -21,16 +21,21 @@
...
     <!--
       TODO:Changing to a PackageReference since we only want to depend on the already GA'ed version of WebPubSub.  Change back after release.
-    <ProjectReference Include="..\..\Azure.Messaging.WebPubSub\src\Azure.Messaging.WebPubSub.csproj" />
-    <ProjectReference Include="..\..\Microsoft.Azure.WebPubSub.Common\src\Microsoft.Azure.WebPubSub.Common.csproj" />
-    -->
+
     <PackageReference Include="Azure.Messaging.WebPubSub" />
     <PackageReference Include="Microsoft.Azure.WebPubSub.Common" />
+        -->
+      <ProjectReference Include="..\..\Azure.Messaging.WebPubSub\src\Azure.Messaging.WebPubSub.csproj" />
```
Comment (JialinXin) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Microsoft.Azure.WebPubSub.AspNetCore.csproj:37]: space issue?

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Microsoft.Azure.WebPubSub.AspNetCore.csproj [hunk 6]
```diff
@@ -21,16 +21,21 @@
 
   <ItemGroup>
     <Compile Include="..\..\Microsoft.Azure.WebPubSub.Common\src\Shared\ConnectionStatesConverter.cs" LinkBase="Shared" />
+    <Compile Include="..\..\Microsoft.Azure.WebPubSub.Common\src\Shared\MqttDisconnectedEventRequestContent.cs" LinkBase="Bindings\Shared" />
+    <Compile Include="..\..\Microsoft.Azure.WebPubSub.Common\src\Shared\MqttConnectEventRequestContent.cs" LinkBase="Bindings\Shared" />
+    <Compile Include="..\..\Microsoft.Azure.WebPubSub.Common\src\Shared\WebPubSubErrorCodeExtensions.cs" LinkBase="Bindings\Shared" />
+    <Compile Include="..\..\\Microsoft.Azure.WebPubSub.Common\src\Shared\MqttConnectCodeToHttpStatusCodeConverter.cs" />
```
Comment (JialinXin) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Microsoft.Azure.WebPubSub.AspNetCore.csproj:27]: there's an additional `\` here.

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,8 @@
 # Release History
 
+## 1.4.0 (2024-09-10)
+- The returned type of `MqttConnectEventRequest.ConnectionContext` and `MqttDisconnectedEventRequest.ConnectionContext` has been changed to `MqttConnectionContext`.
```
Comment (JialinXin) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/CHANGELOG.md:4]: sounds like a breaking change? what's the impact to customer?

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs (hunk 4 above)
Comment (Y-Sindo) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs:42]: It's a feature by design... If users want to handle Web PubSub client and MQTT client `connect` events with unified logic, they can just implement the `OnConnectAsync` method .

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/CHANGELOG.md (hunk 3 above)
Comment (Y-Sindo) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/CHANGELOG.md]: Updated

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Microsoft.Azure.WebPubSub.AspNetCore.csproj (hunk 5 above)
Comment (Y-Sindo) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/Microsoft.Azure.WebPubSub.AspNetCore.csproj]: Fixed

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/CHANGELOG.md (hunk 7 above)
Comment (Y-Sindo) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.Common/CHANGELOG.md]: Yes, it's a breaking change... Now it's reverted.

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs (hunk 4 above)
Comment (JialinXin) [reply sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/src/WebPubSubHub.cs:42]: get it.

File: sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/README.md [hunk 8]
```diff
@@ -109,6 +109,84 @@ private sealed class SampleHub : WebPubSubHub
...
+{
+    internal WebPubSubServiceClient<SampleHub> _serviceClient;
+
+    // Need to ensure service client is injected by call `AddServiceHub<SampleHub2>` in ConfigureServices.
+    public SampleHub2(WebPubSubServiceClient<SampleHub> serviceClient)
+    {
+        _serviceClient = serviceClient;
+    }
+
+    public override ValueTask<WebPubSubEventResponse> OnMqttConnectAsync(MqttConnectEventRequest request, CancellationToken cancellationToken)
```
Comment (vicancy) [inline sdk/webpubsub/Microsoft.Azure.WebPubSub.AspNetCore/README.md:124]: could we still leverage OnConnectAsync for mqtt connections?

**Review by vicancy — APPROVED** [review]

**Pull request merged**